java/arcs/core/util/Scheduler.kt

```diff
@@ -127,7 +129,11 @@ class Scheduler(
 
         // Process agenda
         agenda.forEach { task ->
-            suspendCancellableCoroutine<Unit> { it.resume(task(), timeoutHandler) }
+            suspendCancellableCoroutine<Unit> {
+                log.debug { "Starting $task" }
+                it.resume(task(), timeoutHandler)
```

Comment:
task() throws an exception, does this break the for loop and prevent other items from processing?

Reply:
Yep, and yep :). 

https://pl.kotl.in/Dqr-g9XJw